### ISSUE TITLE
Type not set from original composer.json

### ISSUE THREAD

**Issue by shemgp** [Elendev/nexus-composer-push#2]
Hello,

I wanted to use nexus composer to serve up my packages, but the `type` field doesn't seem to be carried into the repository from the `composer.json`.

I want to use [laravel-module-installer](https://github.com/joshbrw/laravel-module-installer) and it requires that the `type` be `laravel-module`, but when I use this package to push to the reposity I get something like this instead:

```json
{"packages":{"shemgp/theme-module":{"0.0.1":{"name":"shemgp/theme-module","version":"0.0.1","dist":{"url":"https://development.aiias.edu:8081/repository/aiias/shemgp/theme-module/0.0.1/shemgp-theme-module-0.0.1.zip","type":"zip","reference":"4c7335550b1b790cd38d5ef4a028be9c08e72c23","shasum":"4c7335550b1b790cd38d5ef4a028be9c08e72c23"},"time":"2018-12-10T00:35:56+00:00","uid":3250053404},"0.0.2":{"name":"shemgp/theme-module","version":"0.0.2","dist":{"url":"https://development.aiias.edu:8081/repository/aiias/shemgp/theme-module/0.0.2/shemgp-theme-module-0.0.2.zip","type":"zip","reference":"86f1f72da40be3155f66558bdf8f2be85f29161d","shasum":"86f1f72da40be3155f66558bdf8f2be85f29161d"},"time":"2018-12-10T08:42:21+00:00","uid":1381738723}}}}
```

As you can see, the type is `zip` in the `dist` but there's no type in the child fields of the version number so `composer` uses the default, which is `library`.

Thank you for your help.

**Comment by shemgp** [comment]
Sorry I didn't mention it above, but when uploading a zip file, of course I use `curl` to do it, as you have requested.

**Comment by Elendev** [comment]
Thank you for all this investigation.

There is a bug with `0.0.1-alpha2`, I removed the release and I'll investigate why the generated archive doesn't works as expected and fix it.

In the meantime, can you downgrade to 0.0.1-alpha and try with this version ? It might resolve your issue.

I'll come back to you as soon as I've fixed the issue with `0.0.1-alpha2` and the subdirectory of the zip archive.

**Comment by Elendev** [comment]
I've released `v0.0.1.alpha3` that fix the type issue. The `type` is now correctly set (at least on my configuration).

@shemgp can you test on our side and confirm if the issue is fixed ?

**Comment by shemgp** [comment]
Hello,

Yes, I can confirm that with `0.0.1-alpha3` I am able to push to nexus composer repository and the packages have the `type` and not just the `type` but the rest of the fields.

Thank you for your work.
